### PR TITLE
fix: 🐛 Improve multi-version events handling

### PR DESCRIPTION
**Pull request by VictorVicente**
Refactoring to avoid managing the same events on each contract version

BREAKING CHANGE: Y

### CODE REVIEW

**Comment by VictorVicente** [comment]
@monitz87 Could you review?

File: src/contract_wrappers/modules/wallet/vesting_escrow_wallet_wrapper/index.ts [hunk 1]
```diff
@@ -1,12 +1,40 @@
...
+  VestingEscrowWalletEventArgs_3_0_0,
+} from '@polymathnetwork/abi-wrappers';
 import { VestingEscrowWallet_3_0_0, isVestingEscrowWallet_3_0_0 } from './3.0.0';
 import { VestingEscrowWallet_3_1_0, isVestingEscrowWallet_3_1_0 } from './3.1.0';
 
 import Common, { isVestingEscrowWallet } from './common';
-import { ContractVersion, Subscribe, GetLogs } from '../../../../types';
+import { ContractVersion } from '../../../../types';
+
+export const VestingEscrowWalletEvents = {
```
Comment (monitz87) [inline src/contract_wrappers/modules/wallet/vesting_escrow_wallet_wrapper/index.ts]: Why is this necessary if VEW doesn't have changes to events in 3.1?

File: src/contract_wrappers/modules/wallet/vesting_escrow_wallet_wrapper/index.ts [hunk 2]
```diff
@@ -1,12 +1,40 @@
...
+  ...VestingEscrowWalletEvents_3_1_0,
+  ...VestingEscrowWalletEvents_3_0_0,
+};
+export type VestingEscrowWalletEvents = VestingEscrowWalletEvents_3_0_0 | VestingEscrowWalletEvents_3_1_0;
```
Comment (monitz87) [inline src/contract_wrappers/modules/wallet/vesting_escrow_wallet_wrapper/index.ts]: Same here

File: src/contract_wrappers/modules/wallet/vesting_escrow_wallet_wrapper/index.ts (hunk 1 above)
Comment (VictorVicente) [reply src/contract_wrappers/modules/wallet/vesting_escrow_wallet_wrapper/index.ts]: It isn't necessary but I applied it as a good practice. It doesn't harm neither because it only takes 3.0 events, but it we do this always for each new release, it will prevent us to forget some new event.

File: src/contract_wrappers/modules/wallet/vesting_escrow_wallet_wrapper/index.ts (hunk 2 above)
Comment (VictorVicente) [reply src/contract_wrappers/modules/wallet/vesting_escrow_wallet_wrapper/index.ts]: Same answer

File: src/contract_wrappers/modules/transfer_manager/general_transfer_manager_wrapper/index.ts [hunk 3]
```diff
@@ -1,9 +1,32 @@
...
-import { ContractVersion, Subscribe, GetLogs } from '../../../../types';
+import { ContractVersion } from '../../../../types';
+
+export const GeneralTransferManagerEvents = {
+  ...GeneralTransferManagerEvents_3_1_0,
+  ...GeneralTransferManagerEvents_3_0_0,
+};
+export type GeneralTransferManagerEvents = GeneralTransferManagerEvents_3_0_0 | GeneralTransferManagerEvents_3_1_0;
+
+export type GeneralTransferManagerEventArgs = GeneralPermissionManagerEventArgs_3_0_0;
```
[the remark this answers is not in the document]
Comment (VictorVicente) [reply src/contract_wrappers/modules/transfer_manager/general_transfer_manager_wrapper/index.ts]: Good catch! 
I we use the union type of both versions, `GeneralTransferManagerEventArgs` becomes a large set of eventArgs even if they are exactly the same. I think the best approach here is to use the 3.0 eventArgs as a base and add the union with the new ones if there is any, as it is done in both STO modules.

File: src/contract_wrappers/modules/sto/usd_tiered_sto_wrapper/index.ts [hunk 4]
```diff
@@ -1,24 +1,53 @@
...
 
-export type USDTieredSTO = USDTieredSTO_3_0_0 | USDTieredSTO_3_1_0;
+export const USDTieredSTOEvents = {
+  ...USDTieredSTOEvents_3_1_0,
+  ...USDTieredSTOEvents_3_0_0,
+};
+export type USDTieredSTOEvents = USDTieredSTOEvents_3_0_0 | USDTieredSTOEvents_3_1_0;
+
+export type USDTieredSTOEventArgs =
+  | USDTieredSTOEventArgs_3_1_0
```
[the remark this answers is not in the document]
Comment (VictorVicente) [reply src/contract_wrappers/modules/sto/usd_tiered_sto_wrapper/index.ts]: You are right
